internal/pkg/override/cdk.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package override
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+// Executable is the interface to wrap os/exec calls.
+type Executable interface {
+	LookPath(file string) (string, error)
+	Command(name string, args ...string) *exec.Cmd // TODO(efe): Command should set cmd.Dir = rootAbsPath.
+}
+
+// CDK is an Overrider that can transform a CloudFormation template with the Cloud Development Kit.
+type CDK struct {
+	rootAbsPath string // Absolute path to the overrides/ directory.
+
+	out  io.Writer  // Writer for any os/exec calls.
+	fs   afero.Fs   // OS file system.
+	exec Executable // For testing os/exec calls.
+}
+
+// NewCDK instantiates a new CDK Overrider with the file system pointing ot the overrides/ dir.
+func NewCDK(root string, stdout io.Writer, fs afero.Fs, exec Executable) *CDK {
+	return &CDK{
+		rootAbsPath: root,
+		out:         stdout,
+		fs:          fs,
+		exec:        exec,
+	}
+}
+
+// Install runs "npm install" insides the overrides/ directory to pull the latest dependencies.
+func (cdk *CDK) Install() error {
+	if _, err := cdk.exec.LookPath("npm"); err != nil {
+		return &errNPMUnavailable{parent: err}
+	}
+
+	cmd := cdk.exec.Command("npm", "install")
```

Comment:
Is `package.json` a part of the scaffold code that gets generated along with `override.ts`?

Reply:
Yup, see [here](https://github.com/aws/copilot-cli/issues/4208) for a list of files that will be generated part of scaffold.